Client/Frontend/Share/SendToDeviceHelper.swift

```diff
@@ -0,0 +1,73 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Shared
+import Storage
+import SwiftUI
+import UIKit
+
+class SendToDeviceHelper {
+
+    typealias Delegate = InstructionsViewDelegate & DevicePickerViewControllerDelegate
+
+    struct Colors {
+        let defaultBackground: UIColor
+        let textColor: UIColor
+        let iconColor: UIColor
+    }
+
+    enum ViewType {
+        case instructions
+        case picker
+    }
+
+    private var shareItem: ShareItem
+    private var profile: Profile
+    private var colors: Colors
+    private var delegate: Delegate
+
+    init(shareItem: ShareItem, profile: Profile, colors: Colors, delegate: Delegate) {
+        self.shareItem = shareItem
+        self.profile = profile
+        self.colors = colors
+        self.delegate = delegate
+    }
+
+    func initialViewController() -> UIViewController {
+        if !hasAccount() {
+            // Display instructions to log in if user has no account
+            let instructionsView = InstructionsView(backgroundColor: colors.defaultBackground,
+                                                    textColor: colors.textColor,
+                                                    imageColor: colors.iconColor,
+                                                    dismissAction: {
+                self.delegate.dismissInstructionsView()
+            })
+            let hostingViewController = UIHostingController(rootView: instructionsView)
+            #if MOZ_TARGET_SHARETO
+            return hostingViewController
+            #else
+            let navigationController = UINavigationController(rootViewController: hostingViewController)
+            navigationController.modalPresentationStyle = .formSheet
+            return navigationController
+            #endif
```

Comment:
This looks weird to me, but we haven't as a team discussed indentation for pre-processor macros. We can leave it like this, because I think this is what Xcode's indentation would say.... but I couldn't not point this out. lol

Reply:
I think Xcode indents pre-processor macros with no indentation at all if you use the Ctrl+i shortcut. We indent it inline if the code as well in other part of the project. It's kind of annoying since whenever we use the shortcut it goes against what Xcode does, but then I always felt like it looks strange with no indentation at all. Anyway, we can discuss further whenever.

Reply:
Most uses of pre-processor have indentation on the project so I can update this use case and discuss after the team discussion